pkg/customizations/fsnode/fsnode.go

```diff
@@ -1,58 +1,68 @@
 package fsnode
 
 import (
+	"bytes"
+	"encoding/json"
 	"fmt"
 	"os"
 	"path/filepath"
 	"regexp"
+
+	"github.com/osbuild/images/internal/common"
 )
 
 const usernameRegex = `^[A-Za-z0-9_.][A-Za-z0-9_.-]{0,31}$`
 const groupnameRegex = `^[A-Za-z0-9_][A-Za-z0-9_-]{0,31}$`
 
+type baseFsNodeJSON struct {
+	Path  string
+	Mode  *os.FileMode
+	User  interface{}
+	Group interface{}
+}
+
 type baseFsNode struct {
-	path  string
-	mode  *os.FileMode
-	user  interface{}
-	group interface{}
+	baseFsNodeJSON
 }
```

Comment:
I'm having a bit of trouble understanding the reason for this change.  Is it impossible (or messy) to do the unmarshalling into the `File` struct without the nesting?
Could we change the two fsnodes so that they can be unmarshalled unambiguously?  Or perhaps we could have a slightly different serialised format from the internal?

Reply:
Sorry, the commit message should probably be better here. The reason I did it this way is simply that there is already a "File.Path()" function. So just unmarshaling into "File" would require to rename the function and add an exported "File string". I did this in https://github.com/osbuild/images/pull/1497 but the downside of this appraoch is that its now trvial to create invalid node because there is no getters/setters anymore. The original simple map based approach was also not much liked (https://github.com/osbuild/images/pull/1479) and https://github.com/osbuild/images/pull/1500 got the most love so far. But happy to look into more directions here if there are new ideas (fwiw, I can open this one separtely again if you feel it would benefit from a more targeted discussion).

Reply:
Thanks.  The explanation clarifies everything (I didn't connect it to the previous versions and I missed some of the discussion on those PRs).

I think the current solution is fine.  The only alternative I can think of is to basically disconnect the two, so unmarshal into `baseFsNodeJSON` and then copy everything to `baseFsNode` for internal use.  But that's pretty much the same effort and amount of code probably.

Reply:
Yeah, they are probably about the same. I /think/ I even had a version with a `baseFsNodeJSON` and that copied into `baseFsNode` - I don't remember why I decided on this in the end. I don't think go has a really nice pattern for this :/